src/app/app.component.ts

```diff
@@ -23,13 +18,47 @@ export class AppComponent implements OnInit {
   private x_pos: number; // Stores x coordinate of the mouse pointer
   private x_gap: number; // Stores x gap (edge) between mouse and gutter
 
-  constructor(private endpointService: EndpointService,
-              private resourceService: ResourceService,
-              private transmartResourceService: TransmartResourceService,
-              private treeNodeService: TreeNodeService,
-              private constraintService: ConstraintService,
-              private queryService: QueryService,
-              private tableService: TableService) {
+  constructor(private oidcSecurityService: OidcSecurityService,
+              private router: Router) {
+
+    if (this.oidcSecurityService.moduleSetup) {
+      this.onOidcModuleSetup();
+    } else {
+      this.oidcSecurityService.onModuleSetup.subscribe(() => {
+        this.onOidcModuleSetup();
+      });
+    }
+
+    this.oidcSecurityService.onAuthorizationResult.subscribe(
+      (authorizationResult: AuthorizationResult) => {
+        this.onAuthorizationResultComplete(authorizationResult);
+      });
+  }
+
+  private onOidcModuleSetup() {
+    if (window.location.hash) {
+      this.oidcSecurityService.authorizedCallback();
+    } else {
+      if ('/autologin' !== window.location.pathname) {
+        localStorage.setItem('redirect', JSON.stringify(window.location.pathname));
+      }
+
+      this.oidcSecurityService.getIsAuthorized().subscribe((authorized: boolean) => {
+        if (!authorized) {
+          this.router.navigate(['/autologin']);
+        }
+      });
+    }
+  }
+
+  private onAuthorizationResultComplete(authorizationResult: AuthorizationResult) {
+    const path = JSON.parse(localStorage.getItem('redirect'));
+
+    if (authorizationResult === AuthorizationResult.authorized) {
+      this.router.navigate([path]);
+    } else {
+      this.router.navigate(['/']);
+    }
```

Comment:
feels like there auth procedures can be put in 'endpoint service' or 'authentication service' or other names, so that the code is more modularized

Reply:
OK, this is going in "AuthService"

Reply:
maybe give a full name such as AuthenticationService, following the naming convention in glowing bear

Reply:
Will do